lib/src/map/gestures/map_interactive_viewer.dart

```diff
@@ -0,0 +1,268 @@
+import 'package:flutter/gestures.dart';
+import 'package:flutter/widgets.dart';
+import 'package:flutter_map/flutter_map.dart';
+import 'package:flutter_map/src/map/gestures/services/base_services.dart';
+
+class MapInteractiveViewer extends StatefulWidget {
+  final ChildBuilder builder;
+  final MapControllerImpl controller;
+
+  const MapInteractiveViewer({
+    super.key,
+    required this.builder,
+    required this.controller,
+  });
+
+  @override
+  State<MapInteractiveViewer> createState() => MapInteractiveViewerState();
+}
+
+class MapInteractiveViewerState extends State<MapInteractiveViewer>
+    with TickerProviderStateMixin {
+  TapGestureService? _tap;
+  LongPressGestureService? _longPress;
+  SecondaryTapGestureService? _secondaryTap;
+  SecondaryLongPressGestureService? _secondaryLongPress;
+  TertiaryTapGestureService? _tertiaryTap;
+  TertiaryLongPressGestureService? _tertiaryLongPress;
+  DoubleTapGestureService? _doubleTap;
+  ScrollWheelZoomGestureService? _scrollWheelZoom;
+  TwoFingerGesturesService? _twoFingerInput;
+  DragGestureService? _drag;
+  DoubleTapDragZoomGestureService? _doubleTapDragZoom;
+  CtrlDragRotateGestureService? _ctrlDragRotate;
+
+  MapCamera get _camera => widget.controller.camera;
+
+  MapOptions get _options => widget.controller.options;
+
+  InteractionOptions get _interactionOptions => _options.interactionOptions;
+
+  @override
+  void initState() {
+    super.initState();
+    widget.controller.interactiveViewerState = this;
+    widget.controller.addListener(reload);
+
+    // callback gestures for the application
```

Comment:
The problem is that most of the gestures have differnt parameters. If we use only one callback or a stream the user needs to cast the event type and we need additional classes to wrap all parameters together. We can go that way but I'm not sure if it would make things easier for us and for the users.

Reply:
We can remove those callbacks if you want. Currently all gestures will get emitted by the event system anyway. 
I personally think that callbacks are easier to understand and implement for users and the their implementation overhead seems quite low to me. It's probably a question of how beginner friendly we want our API to be.

Reply:
I'm just looking at what we've done with the polyline hit detection. It's perhaps not quite as simple as a callback, but it is the most flexible option by far, and not too much more difficult to implement, especially with some helpful documentation and demos.
What do others think?